### ISSUE TITLE
#4162 Regressions

### ISSUE THREAD

**Issue by Kravickas** [RPCS3/rpcs3#4247]
Regressions caused by #4162 / commit f862fe4

**Fight Night Round 4** 
Game is no longer ingame - while loading ingame -> 
`F {PPU[0x1000010] Thread (LoopSimulation) [0x0007a968]} MEM: Access violation reading location 0xa69d1ecc
`
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1778084/RPCS3.log.gz)

__________________________
**NHL 13/15/16** 
Games are no longer ingame - while loading ingame -> 
`F {PPU[0x1000006] Thread (NHL Render Thread) [0x009733dc]} MEM: Access violation writing location 0x9017e7e0`
`F {SPU[0x2000002] Thread (Job Manager - Job Thread)} class std::runtime_error thrown: Verification failed: (in file **Emu\Cell\SPUThread.cpp:966)`
or 
`F {PPU[0x1000006] Thread (NHL Render Thread) [0x009034c8]} MEM: Access violation writing location 0xff122e59`
or
`
F {SPU[0x2000005] Thread (Job Manager - Job Thread)} class std::runtime_error thrown: Halt
(in file Emu\Cell\SPUThread.cpp:1863)
`
and forever loading after resume
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1800564/RPCS3.log.gz)

**Comment by greggameplayer** [comment]
I think it's because he change a little the way that the OPCode is implemented.

**Comment by Kravickas** [comment]
Found another game regressed almost the same, so changing title to general pr regressions.

**Comment by Kravickas** [comment]
first commit 6954dd84 introduced:
 `E {rsx::thread} RSX: nv406e::semaphore_acquire has timed out. semaphore_address=0x403004B0` 
in NHL while loading to ingame 

and  `F {SPU[0x2000002] Thread (Job Manager - Job Thread)} class std::runtime_error thrown: Verification failed: 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\../Common/texture_cache.h:193)
` which was i guess fixed somewhere between these commits

 but

commit f862fe45 bringed Fatal errors regression in 1. post
_________
So we have timeouts and then access viol. / Halt fatal errors after f862fe45

**Comment by Asinin3** [comment]
After reading some of the above comments, i double checked to see if this was the PR that regressed Yakuza 5 and 0 (Ishin as well according to a user on discord). And I can confirm that that it caused these games to spam semaphore timeout messages and run at 1fps. Reaching ingame takes around half an hour after this change.

**Comment by stride21** [comment]
Shadow of The Colossus in The Ico & Shadow of the Colossus Collection no long goes ingame and has the same semaphore timeout messages. 

[RPCS3.7z](https://forums.rpcs3.net/attachment.php?aid=5176)

**Comment by stride21** [comment]
PR #4384 fixed The ICO & Shadow of the Colossus Collection for me.

**Comment by kroaton** [comment]
Fight Night Round 4 is still broken after #4384 (rpcs3-v0.0.5-6647-64992f75_win64) with the same "MEM: Access violation reading location 0xa6a7b414" problem caused by  #4162

[log.txt](https://github.com/RPCS3/rpcs3/files/1887782/log.txt)

rpcs3-v0.0.5-6592-a36acf49_win64 (pre #4162) is the last build where it works properly.

**Comment by kroaton** [comment]
#4411 #4409 #4402 #4352 - all of these seem to be related to this MEM access problem

@kd-11

**Comment by cooperate** [comment]
I was directed to this thread for this error.  On the latest build I am still getting this error:
`F {SPU[0x2000005] Thread (Job Manager - Job Thread)} class std::runtime_error thrown: Halt
(in file Emu\Cell\SPUThread.cpp:2337)
F {SPU[0x2000002] Thread (Job Manager - Job Thread)} class std::runtime_error thrown: Halt
(in file Emu\Cell\SPUThread.cpp:2337)`

This is on NHL 3-on-3 Arcade NPUB30058
It crashes right before going in-game (during loading screen).
On build rpcs3-v0.0.5-6652-41f4bb90_win64

**Comment by JMC47** [comment]
I'll add NHL 09 (BLUS30168) (I'm guessing they all use the same engine?) to the list of games with this issue.

**Comment by cooperate** [comment]
As far as I know every NHL game is suffering from this issue yes.

**Comment by kroaton** [comment]
Fight Night Round 4 has regressed again in the latest build...

Same MEM:Access violation problem 

Somewhere between #4424 and #4419

**Comment by cooperate** [comment]
@JMC47 I can confirm that Null Rederer will get 3 on 3 NHL Arcade (NPUB30058) In-Game using the null render.

Using RPCS3 latest build as of this posting date.

**Comment by Nicba1010** [comment]
Sly Cooper Thieves in time also regresses NPEA00429
Spams semaphore lock timeouts and runs at from 1-8 fps, sometimes it manages to push more frames out. Without this issue the game would and I say this with 99% certainty be playable.

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1977536/RPCS3.log.gz)

**Comment by Kravickas** [comment]
Loading to Ingame is broken after commit f862fe4 in:
NHL 12, 13, 14, 15, 16
NHL 3-on-3 Arcade
Fight night round 4
Fight night round champions

all games get access violation

**Comment by Nicba1010** [comment]
semaphore locks for NPEA00429 still persisting on master

**Comment by Kravickas** [comment]
well, NHL game kinda works, no access violation but can sometimes hang on loading to ingame (mostly because not builded spu cache (try it few times))

Fight Night 4/Champion need Cpu blit to bypass access violation

**Comment by Asinin3** [comment]
Yakuza 5, 0 and Dead Souls are no longer spamming semaphore errors.
Sly Cooper Thieves in time is still spamming the error, but my fps is still pretty normal. 
![image](https://user-images.githubusercontent.com/15137035/41857987-45abd5d0-7888-11e8-8d98-8d9747724261.png)
And before you ask, this is with the default timeout value in config.yml

**Comment by kd-11** [comment]
If any issues remain, let me know so we can reopen this.

**Comment by kroaton** [comment]
Fight Night 4 is still suffering from this problem in the latest builds.

The game used to be fully playable (apart from broken audio and occasional stutter) in #4424.

Force Cpu Blit does indeed bypass the memory access violation but the player models are gone now (it might be some other regression or side effect of Force Cpu Blit).

**Comment by kd-11** [comment]
Issue is under investigation.

**Comment by Lozs** [comment]
Tears to Tiara 2 keeps having the same issue of crashing almost everywhere.

**Comment by kd-11** [comment]
Is there a special way to make Fight Night crash? I'm using a build from https://github.com/RPCS3/rpcs3/pull/5099
![image](https://user-images.githubusercontent.com/15904127/45294457-1284f480-b504-11e8-87e1-6e64b37f9604.png)

**Comment by kd-11** [comment]
Looks like that PR is what fixed it. I tried an older build and it crashes when going ingame. I guess the reported regressions now need to be retested.
Issues possibly fixed by https://github.com/RPCS3/rpcs3/pull/5099

**Comment by Kravickas** [comment]
FNR4 still F {PPU[0x1000010] Thread (LoopSimulation) [0x0007a968]} MEM: Access violation reading location 0xb5bb4b9c without CPU Blit

With CPU Blit both FN4 and FNCH go ingame

**Comment by kd-11** [comment]
Which build are you using? It consistently works fine for me with my local build.

**Comment by Kravickas** [comment]
Latest Master

**Comment by kd-11** [comment]
Not master, build from PR https://github.com/RPCS3/rpcs3/pull/5099. Its not yet merged.

**Comment by Kravickas** [comment]
Sorry I tho i was reading that maybe another PR fixed it so i tested master...

Latest build from #5099 work fine without CPU Blit

**Comment by kroaton** [comment]
@kd-11 @Kravickas 


When trying to boot BLUS30608 - Fight Night Champion, I still get "F {PPU[0x100001a] Thread (MoviePlayer2 Decode Thread) [0x000e2bd4]} MEM: Access violation reading location 0x0" using #5099 without CPU Blit turned on.

Am I missing something?

**Comment by kd-11** [comment]
Yes, that is a different bug, likely a race condition. Look at the 'address' there, its 0 while the others have actual address values but were crashing incorrectly. You should try with ruipin's PR https://github.com/RPCS3/rpcs3/pull/5115.
